### ISSUE TITLE
4 card badugi not recognized

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#174]
My 4-card was called a 3 card and Gerry's 3-card was calle a 2 card. WTF? I thought we've played enough Badugi to not have this issue. Could I have introduced it with my recent fix?

```
               My cards are: [6c] [Qs] [9d] [6h]
gerdog    3h Ad 2h 2s  Two Card: 2 A     2h Ad 3h 2s
deadhead  6c Qs 9d 6h  Three Card: Q 9 6 Qs 9d 6c 6h
deadhead wins 16900
```
 
FWIW, it should be easy to reproduce and easy to debug, but still...

**Comment by ctm** [comment]
Oops. Mine was called correctly (my sixes were duplicated), but gerdog's wasn't. His should have been a 3 card 3 2 A.

**Comment by ctm** [comment]
When I wrote my badugi evaluators, I didn't write a test that exhaustively tests the problem space, like I did for my regular high hand evaluator. On top of that, I also didn't write the obvious recursive solution, instead I went directly with a bit-fiddly one that was tricky.

A bug like this is completely unacceptable, so not only will I add a test case for Gerry's specific hand, but I'll also write the recursive solution and exhaustively test that against my bit fiddler. That means it will take slightly longer before this is fixed, but it will still be done by tonight's tournament. It does mean a slightly longer delay for #86, but I'm still hopeful that I'll be able to use that code on Saturday, when I won't be around to start the tourney.

**Comment by ctm** [comment]
I'm half way (for some definitions of half) there, in that I added a test case for Gerry's hand and made sure that it failed in the same way it failed last night. I then went through and identified the bug that was in my bit fiddling code. FWIW, my comments were basically correct, but there was something I needed to test that I didn't.  I've since fixed that and now the new test case passes.

However, I haven't yet done the exhaustive testing, nor written the simpler recursive evaluator that I'll use to compare my bit-fiddling evaluator against. So, even though I'm deploying the bug fix I just added, I'm not closing this ticket (nor am I moving on to work on #86, yet).
